server/src/main/java/org/opensearch/index/ShardIndexingPressure.java

```diff
@@ -0,0 +1,439 @@
+
+        for (Map.Entry<Long, ShardIndexingPressureTracker> shardEntry :
+                this.shardIndexingPressureStore.getShardIndexingPressureHotStore().entrySet()) {
+            IndexingPressurePerShardStats shardStats = new IndexingPressurePerShardStats(shardEntry.getValue(),
+                    isEnforcedMode);
+            statsPerShard.put(shardEntry.getKey(), shardStats);
+        }
+        return new ShardIndexingPressureStats(statsPerShard, memoryManager.totalNodeLimitsBreachedRejections.get(),
+            memoryManager.totalLastSuccessfulRequestLimitsBreachedRejections.get(),
+            memoryManager.totalThroughputDegradationLimitsBreachedRejections.get(),
+            shardIndexingPressureSettings.isShardIndexingPressureEnabled(),
+            isEnforcedMode);
+    }
+
+    ShardIndexingPressureStats coldStats() {
+        Map<Long, IndexingPressurePerShardStats> statsPerShard = new HashMap<>();
+        boolean isEnforcedMode = shardIndexingPressureSettings.isShardIndexingPressureEnforced();
+
+        for (Map.Entry<Long, ShardIndexingPressureTracker> shardEntry :
+                this.shardIndexingPressureStore.getShardIndexingPressureColdStore().entrySet()) {
+            IndexingPressurePerShardStats shardStats = new IndexingPressurePerShardStats(shardEntry.getValue(),
+                    isEnforcedMode);
+            statsPerShard.put(shardEntry.getKey(), shardStats);
+        }
+        return new ShardIndexingPressureStats(statsPerShard, memoryManager.totalNodeLimitsBreachedRejections.get(),
+            memoryManager.totalLastSuccessfulRequestLimitsBreachedRejections.get(),
+            memoryManager.totalThroughputDegradationLimitsBreachedRejections.get(),
+            shardIndexingPressureSettings.isShardIndexingPressureEnabled(),
+            isEnforcedMode);
+    }
+
+    ShardIndexingPressureStats topStats() {
+        return new ShardIndexingPressureStats(Collections.emptyMap(), memoryManager.totalNodeLimitsBreachedRejections.get(),
+            memoryManager.totalLastSuccessfulRequestLimitsBreachedRejections.get(),
+            memoryManager.totalThroughputDegradationLimitsBreachedRejections.get(),
+            shardIndexingPressureSettings.isShardIndexingPressureEnabled(),
+            shardIndexingPressureSettings.isShardIndexingPressureEnforced());
+    }
+
+    ShardIndexingPressureTracker getShardIndexingPressureTracker(ShardId shardId) {
+        return shardIndexingPressureStore.getShardIndexingPressureTracker(shardId);
+    }
+
+    public static boolean isShardIndexingPressureAttributeEnabled() {
+        Iterator<DiscoveryNode> nodes = clusterService.state().getNodes().getNodes().valuesIt();
+        while (nodes.hasNext()) {
+            if (Boolean.parseBoolean(nodes.next().getAttributes().get(SHARD_INDEXING_PRESSURE_ENABLED_ATTRIBUTE_KEY)) == false) {
+                return false;
+            }
+        }
+        return true;
+    }
```

Comment:
Let's move this to `ShardIndexingPressureSettings`

Reply:
makes sense. Moved to `ShardIndexingPressureSettings` for containment of logic.